internal/commands/user/delete/command.go

```diff
@@ -0,0 +1,164 @@
+package delete
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/commands/user/shared"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/10gen/realm-cli/internal/utils/flags"
+	"github.com/spf13/pflag"
+)
+
+// Command is the `user delete` command
+var Command = cli.CommandDefinition{
+	Use:         "delete",
+	Description: "Delete the user(s) from a Realm application",
+	Help:        "user delete",
+	Command:     &command{},
+}
+
+type command struct {
+	inputs      inputs
+	outputs     []output
+	realmClient realm.Client
+}
+
+type output struct {
+	user realm.User
+	err  error
+}
+
+func (cmd *command) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+
+	fs.StringSliceVarP(&cmd.inputs.Users, flagUsers, flagUsersShort, []string{}, flagUsersUsage)
+	fs.VarP(
+		flags.NewEnumSet(&cmd.inputs.ProviderTypes, shared.ValidProviderTypes),
+		shared.FlagProvider,
+		shared.FlagProviderShort,
+		shared.FlagProviderUsage,
+	)
+	fs.Lookup(shared.FlagProvider).NoOptDefVal = shared.ProviderTypeInteractive
+	fs.VarP(&cmd.inputs.State, shared.FlagStateType, shared.FlagStateTypeShort, shared.FlagStateTypeUsage)
+	fs.Lookup(shared.FlagProvider).NoOptDefVal = shared.ProviderTypeInteractive
+	fs.VarP(&cmd.inputs.Status, shared.FlagStatusType, shared.FlagStatusTypeShort, shared.FlagStatusTypeUsage)
+	fs.Lookup(shared.FlagStatusType).NoOptDefVal = shared.StatusTypeInteractive.String()
+}
+
+func (cmd *command) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+func (cmd *command) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = realm.NewAuthClient(profile.RealmBaseURL(), profile.Session())
+	return nil
+}
+
+func (cmd *command) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, err := cli.ResolveApp(ui, cmd.realmClient, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	var users []realm.User
+	if len(cmd.inputs.Users) < 1 {
```

Comment:
[nit] let's have the check be `== 0`

[thought] could we push this logic into the `inputs` code, something like `users, usersErr := cmd.inputs.ResolveUsers(...)`?

Reply:
I'll move it